core/templates/dev/head/pages/admin-page/services/admin-config-tab-backend-api.service.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
```

Comment:
Please add tests for this file.

Reply:
I am working on this and am trying to find what the http requests return. Where could I find help/documentation regarding this?

Reply:
As in you are trying to find what the backend returns? If so, here is what you can do..
- Find the URL the requests are made to
- Find the corresponding handler in main.py
- See the python code in the handler to view what the data returned will contain (sometimes looking at the corresponding test file might be a better idea).

Reply:
https://github.com/oppia/oppia/blob/ac647958462b39ed0a565ce1c8e35bad08719331/core/controllers/admin.py#L147-L241
In the above function, for `revert_config_property`, `{}` is returned.
But when the following test file is executed, the test fails with `Error: No pending request to flush !`.

```
// TODO(#7222): Remove the following block of unnecessary imports once
// the code corresponding to the spec is upgraded to Angular 8.
import { UpgradedServices } from 'services/UpgradedServices';
// ^^^ This block is to be removed.
require('pages/admin-page/services/admin-config-tab-backend-api.service');

describe('Admin Config Tab Backend API service', function() {
  let AdminConfigTabBackendApiService = null;
  let sampleDataResult = null;
  let $rootScope = null;
  let $scope = null;
  let $httpBackend = null;

  beforeEach(angular.mock.module('oppia'));
  beforeEach(angular.mock.module('oppia', function($provide) {
    let ugs = new UpgradedServices();
    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
      $provide.value(key, value);
    }
  }));

  beforeEach(angular.mock.inject(function($injector) {
    AdminConfigTabBackendApiService = $injector.get(
      'AdminConfigTabBackendApiService');
    $rootScope = $injector.get('$rootScope');
    $scope = $rootScope.$new();
    $httpBackend = $injector.get('$httpBackend');
  }));

  afterEach(function() {
    $httpBackend.verifyNoOutstandingExpectation();
    $httpBackend.verifyNoOutstandingRequest();
  });

  it('should revert specified config property to default value',
    function() {
      let successHandler = jasmine.createSpy('success');
      let failHandler = jasmine.createSpy('fail');

      $httpBackend.expect('POST', '/adminhandler').respond(200, {});
      AdminConfigTabBackendApiService.revertConfigProperty('promo_bar_enabled').then(
        successHandler, failHandler);
      $httpBackend.flush();

      expect(successHandler).toHaveBeenCalled();
      expect(failHandler).not.toHaveBeenCalled();
    }
  );
});

```

Reply:
done